src/nvim/event/socket.c

```diff
@@ -90,14 +71,38 @@ int socket_watcher_start(SocketWatcher *watcher, int backlog, socket_cb cb)
   int result;
 
   if (watcher->stream->type == UV_TCP) {
-    result = uv_tcp_bind(&watcher->uv.tcp.handle,
-                         (const struct sockaddr *)&watcher->uv.tcp.addr, 0);
+    struct addrinfo *ai = watcher->uv.tcp.addrinfo;
+
+    for (; ai; ai = ai->ai_next) {
+      memcpy(&watcher->uv.tcp.addr, ai->ai_addr, ai->ai_addrlen);
+      result = uv_tcp_bind(&watcher->uv.tcp.handle,
+                           (const struct sockaddr *)&watcher->uv.tcp.addr, 0);
```

Comment:
Right, I guess it would be more explicit to use `watcher` since it's also used for all the following calls.. but using `ai->ai_addr` I can avoid the cast and it just looks better.

:+1: